### DISCUSSION TITLE
Why is this converted to an an array then immediately back into a list

### DISCUSSION THREAD

**Discussion by sebastianwhiffen** [microsoft/RulesEngine#589] [Q&A]
Hey all, 

Was just looking through this code to gain a greater understanding of the fantastic tool you've created. 

Noticed this:
![image](https://github.com/microsoft/RulesEngine/assets/107291523/a6cf3f96-c29a-4dcf-9591-660be6f706fe)

is this intentional? if so what is the reasoning behind it

**Comment by asulwer** [comment]
the name of the RuleParameter input has a number added to its name.  its closely tied to its string name.  i am working on removing this tight coupling

**Reply by abbasc52** [reply]
The sortedRulesParam was a contribution to ensure changed ordering of rule param does not impact output caching.

Although, yes array could be sorted without converting to list.

**Reply by asulwer** [reply]
as to the sorted question,  i implemented it like this which prevents that array from being converted to a list and then back to an array

`Array.Sort(inputs, (RuleParameter a, RuleParameter b) => string.Compare(a.Name, b.Name));`

maybe i misunderstood, but i gathered the question was why convert from Array -> List -> Array.  the original parameter is an object[] which is converted to a list and back to an array. i am working on solving this.  for now the first part i stated above prevents extra conversion
